### PR TITLE
Katleiah/custom costumes prompts

### PR DESCRIPTION
**Pull request by katleiahramos**
This PR fixes the issue of customized or drawn sprite images note showing when selected as a prompt option. 

Before:
image

Changes in this PR:
image

## Links
[Jira Ticket](https://codedotorg.atlassian.net/browse/STAR-2185) 

## Follow-up work
This fix requires populating source urls for custom animations in the `SpritelabInput` component. It would be great to see if there is a way to centralize how and when we fetch and generate these urls, rather than having to generate them in a duplicate way in multiple locations in the code. A further investigation into the ways in which we rely on `sourceUrl` to be null for custom animations is needed. 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by katleiahramos** [comment]
One thing I did notice is that (because of the way we have to fetch the animation URL from the API) there is sometimes a lag of rendering the custom sprite image when running and resetting the play space. Given our current setup, and the fact that even default sprites have this behavior, I'm not sure if we should try and address this as it's not blocking functionality. Open to ideas though if others have them! 


https://user-images.githubusercontent.com/35442460/161820868-ab14c935-bf7b-4226-83a1-ea94278d3b16.mov

File: apps/src/p5lab/spritelab/SpritelabInput.jsx [hunk 1]
```diff
@@ -6,10 +6,11 @@ import * as shapes from '../shapes';
 import {KeyCodes} from '@cdo/apps/constants';
 import {selectors} from '@cdo/apps/lib/tools/jsdebugger/redux';
 import {PromptType} from '../redux/spritelabInput';
-
+import {animationSourceUrl} from '../redux/animationList';
```
Comment (epeach) [inline apps/src/p5lab/spritelab/SpritelabInput.jsx:9]: Nit: Can we add a blank line here to separate the imports from the class? Makes it zone in when reading.

File: apps/src/p5lab/spritelab/SpritelabInput.jsx [hunk 2]
```diff
@@ -59,9 +60,16 @@ class SpritelabInput extends React.Component {
...
-    );
+
+    Object.entries(animationPropsByKey).forEach(animation => {
+      const animationKey = animation[0];
+      const animationData = animation[1];
+      const url =
+        animationData.sourceUrl ||
+        // If custom animation, generate URL from animations API
+        animationSourceUrl(animationKey, animationData, this.props.channelId);
+      spriteMap[`image_${animation[1].name}`] = url;
```
Comment (epeach) [inline apps/src/p5lab/spritelab/SpritelabInput.jsx]: For consistency, since we've named `animation[1]` as `animationData`, I think we can use that here.

Comment (katleiahramos) [reply apps/src/p5lab/spritelab/SpritelabInput.jsx]: ah great catch! totally forgot to actually use the variable I made haha!

Comment (katleiahramos) [reply apps/src/p5lab/spritelab/SpritelabInput.jsx]: updated: https://github.com/code-dot-org/code-dot-org/pull/45636/commits/de2e752831832b88479b4ca3eb55337de03ec365

File: apps/src/p5lab/spritelab/SpritelabInput.jsx (hunk 1 above)
Comment (katleiahramos) [reply apps/src/p5lab/spritelab/SpritelabInput.jsx:9]: updated! https://github.com/code-dot-org/code-dot-org/pull/45636/commits/de2e752831832b88479b4ca3eb55337de03ec365

**Review by epeach — APPROVED** [review]
:+1:

**Pull request merged**